src/main/java/roomescape/api/reservations/controller/ReservationsController.java

```diff
@@ -0,0 +1,53 @@
+package roomescape.api.reservations.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import roomescape.api.reservations.dto.request.ReservationRegisterRequest;
+import roomescape.api.reservations.service.ReservationsService;
+import roomescape.db.entity.ReservationsEntity;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+
+@RestController
+@RequestMapping("/reservations")
+@RequiredArgsConstructor
+public class ReservationsController {
+
+    private final ReservationsService reservationsService;
+
+    private List<ReservationsEntity> reservationEntities = new ArrayList<>();
+    private AtomicLong index = new AtomicLong();
+
+    @GetMapping
+    public ResponseEntity<List<ReservationsEntity>> getReservations() {
+        return ResponseEntity.status(HttpStatus.OK).body(reservationsService.getReservations());
+    }
+
+    @PostMapping
+    public ResponseEntity<Void> createReservation(
+            @RequestBody @Valid final ReservationRegisterRequest reservationRegisterRequest
+    ) {
+
+        return ResponseEntity.status(HttpStatus.CREATED)
+                .header("Location", URI.create("/reservations/" +
+                        reservationsService.createReservations(reservationRegisterRequest.name(), reservationRegisterRequest.date(), reservationRegisterRequest.time())).toString()
```

Comment:
Service 계층 호출이 헤더 지정 내부에서 이루어지네요.
이 핸들러가 어떤 역할을 수행하는지 파악하기 어렵지 않나요?

`ResponseEntity`가 제공하는 `created`를 사용해서 더 개선할 수 있을 것 같아요.
```suggestion
long reservationsId = reservationsService.createReservations(
    reservationRegisterRequest.name(),
    reservationRegisterRequest.date(),
    reservationRegisterRequest.time()
);
return ResponseEntity.created(URI.create("/reservations/" + reservations)).build();
```

Reply:
맞습니다..! 따로 빼서 수정해보도록 하겠습니다!! 감사합니다 : ) 다만 ResponseEntity에서 status를 사용하는 이유는 의미를 분명하게 하려고 했던 제 개인 취향이었습니다..ㅎㅎ